bot/kodiak/tests/evaluation/test_check_runs.py

```diff
@@ -562,3 +567,38 @@ async def test_mergeable_update_always_no_require_automerge_label_missing_label(
     assert api.queue_for_merge.call_count == 0
     assert api.merge.call_count == 0
     assert api.dequeue.call_count == 0
+
+
+@pytest.mark.asyncio
+async def test_duplicate_check_suites() -> None:
+    """
+    Kodiak should only consider the most recent check run when evaluating a PR
+    for merging.
+
+    Regression test.
+    """
+    api = create_api()
+    pull_request = create_pull_request()
+    pull_request.mergeStateStatus = MergeStateStatus.BEHIND
+    branch_protection = create_branch_protection()
+    branch_protection.requiresStatusChecks = True
+    branch_protection.requiredStatusCheckContexts = ["Pre-merge checks"]
+    mergeable = create_mergeable()
+    await mergeable(
+        api=api,
+        pull_request=pull_request,
+        branch_protection=branch_protection,
+        check_runs=[
+            create_check_run(
+                name="Pre-merge checks", conclusion=CheckConclusionState.NEUTRAL
+            ),
+            create_check_run(
+                name="Pre-merge checks", conclusion=CheckConclusionState.FAILURE
+            ),
+            create_check_run(
+                name="Pre-merge checks", conclusion=CheckConclusionState.SUCCESS
```

Comment:
does the GitHub API order the results with the most recent ones last?

Reply:
I opened a support ticket about that but my assumption here is yes